.github/workflows/build-examples.yml

```diff
@@ -0,0 +1,57 @@
+name: Build Library Examples
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        fqbn: [
+          "arduino:samd:mkrwifi1010",
+          "arduino:samd:nano_33_iot",
+          "arduino:megaavr:uno2018:mode=on",
+          "arduino:mbed:nano33ble"
+        ]
+
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+
+      - name: Setup arduino-cli
+        env:
+          VERSION: latest
+        run: |
+          set -x
+          wget -P $HOME https://downloads.arduino.cc/arduino-cli/arduino-cli_${VERSION}_Linux_64bit.tar.gz
+          mkdir $HOME/bin
+          tar xf $HOME/arduino-cli_${VERSION}_Linux_64bit.tar.gz  -C $HOME/bin
+          echo ::add-path::$HOME/bin
```

Comment:
Why don't use the [official setup-arduino-cli github action](https://github.com/arduino/actions/tree/master/setup-arduino-cli) to remove a bit of boilerplate? In #32 You can see the action working.

Reply:
I agree on using the official GitHub action. Everything else looks perfectly good to me. Well done!

Reply:
Is there a way we can slim the GitHub action down? It seems overcomplicated internally (depends on Node.js) etc.

Reply:
As we discussed the nodejs code gives more flexibility (i.e. pinning only the minor version for the cli) and the chance to leverage other actions code as modules. Additionally the action code is testable with unit testing. KISS is always to prefer, but in case you discover  need something more powerful, we'll be happy to help :smiley_cat: